Doc/library/codecs.rst

```diff
@@ -242,6 +242,13 @@ wider range of codecs when working with binary files:
    :func:`iterencode`.
 
 
+.. function:: escape_decode(data, errors=None)
+
+   Decode the bytes-like object *data* and return a tuple (decoded object,
+   length consumed). This is useful for decoding ascii escape sequences mixed
+   with unicode characters.
```

Comment:
What is "mixed with unicode characters" supposed mean in this context?  data is a bytes-like object, it can't contain unicode runes.  We should include an example of what this does that is different than using one of the text encoding codecs.

Reply:
As Matthieu Dartiailh described on bugs.python.org, an example of ascii decode characters mixed with unicode is `'Δ\nΔ'`. 

Here is the difference:

``` python
>>> codecs.unicode_escape_decode('Δ\nΔ')
('Î\x94\nÎ\x94', 5)
>>> codecs.escape_decode('Δ\nΔ')
(b'\xce\x94\n\xce\x94', 5)
>>> codecs.escape_decode('Δ\nΔ')[0].decode('utf-8')
'Δ\nΔ'
```

Reply:
A [real-world example](https://stackoverflow.com/questions/70457699/decode-individual-octal-characters-in-string-variable). We can assume that many more homegrown 'solutions' exist. 

Wouldn't it be great if kludgy, slow, error-prone workarounds people have come up with were replaced with something elegant and Python-worthy?

Please consider that this function is so rarely seen outside the Python developer world *because it is kept almost a secret.*